soc/riscv32/openisa_rv32m1/Kconfig.defconfig

```diff
@@ -29,6 +29,7 @@ config XIP
 
 config RISCV_GENERIC_TOOLCHAIN
 	bool
+	default y if "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
 	default n
```

Comment:
Another variant:

```
config RISCV_GENERIC_TOOLCHAIN
	bool
	default "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
```

Or this:

```
config RISCV_GENERIC_TOOLCHAIN
	def_bool "$(ZEPHYR_TOOLCHAIN_VARIANT)" = "zephyr"
```

The value for `default`, etc., can be any expression.